Tools/WinMLRunner/BindingUtilities.h

```diff
@@ -144,6 +144,35 @@ namespace BindingUtilities
         return elementStrings;
     }
 
+    template <typename T>
+    void PreProcessImageToBinding(const SoftwareBitmap& softwareBitmap, ModelBinding<T>& binding, float scale, const std::array<float, 3>& meanStdDev)
+    {
+        const auto imgHeight = softwareBitmap.PixelHeight();
+        const auto imgWidth = softwareBitmap.PixelWidth();
+        const auto resultSize = imgHeight * imgWidth * 3;
+        if (binding.GetDataBufferSize() != resultSize)
+        {
+            throw hresult_invalid_argument(L"Image Input size/shape is different from what model expects");
+        }
+
+        T* bindingData = binding.GetData();
+        if (scale != 1.0f &&
+            (meanStdDev[0] != 0 || meanStdDev[1] != 0 || meanStdDev[2] != 0))
+        {
+            Buffer sbBuffer(imgHeight * imgWidth * 4);
```

Comment:
This code is doing 3 copies into 4 buffers. SoftwareBitmap->Buffer->ModelBinding<T>->Tensor*.

The first copy can be eliminated if you use LockBuffer on SoftwareBitmap to access the underlying data.
The last copy can be eliminated if you use ITensorNative to write the transformed data directly into the Tensor* backing memory.

Reply:
You can take a look at the code in function `SoftwareBitmapToSoftwareTensor()` at [here](https://github.com/Microsoft/Windows-Machine-Learning/blob/master/Samples/CustomTensorization/CustomTensorization/TensorConvertor.cpp)

Reply:
Great, thanks for the pointer!  In trying to keep the code as simple as possible, does a VectorView of the Tensor create a copy?  I'm asking because the conversion to float16 using the byte interface will require a specialized template function with a manual conversion.  I would rather keep this in the VectorView<float> to Tensor<Float16> code (I'm assuming it will happen automatically)?